### PR TITLE
PHPUnit Cleanup - Fixes #2053

### LINKED ISSUE #2053: PHPUnit Doesn't Clear Caches
Before submitting your issue, please make sure that you've checked the checkboxes below.

- [x] I am running the [latest release](https://github.com/CachetHQ/Cachet/releases/latest) version of Cachet.
- [x] I am running at least PHP 5.5.9. *You can check this by running `php -v`.*
- [x] I have ran `rm -rf bootstrap/cache/*`.

### Expected behaviour

After running `phpunit`, I expect my environment to not be mutated.

### Actual behaviour

PHPUnit bootstraps it's environment and never clears it's cache once completed (application stays in testing).

### Steps to reproduce

1. Have a default installation (pre-setup and seeded)
2. Run `vendor/bin/phpunit`
3. Go to your site, notice you're now on `/setup`
4. Run `artisan env` and notice you're now in `testing`

### PR DESCRIPTION
**Pull request by ConnorVG**
See #2053

### CODE REVIEW

**Comment by ConnorVG** [comment]
Sorry, had to make a cup of tea :trollface:

**Comment by GrahamCampbell** [comment]
image

**Comment by GrahamCampbell** [comment]
Thanks, looks good. Will merge once travis is done with it.

**Comment by ConnorVG** [comment]
Trust me, I tested it locally.

It totally works :trollface: 


:trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface: :trollface:

**Comment by GrahamCampbell** [comment]
:trollface:

File: tests/TestListener.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @return void
+     */
+    public function endTestSuite(PHPUnit_Framework_TestSuite $suite)
+    {
+        if ($suite->getName() !== 'Cachet Test Suite') {
+            return;
+        }
+
+        foreach (glob(__DIR__.'/../bootstrap/cache{,t}/*.php', GLOB_BRACE) as $file) {
```
Comment (yoyosan) [inline tests/TestListener.php]: That's a neat trick. :+1:

Comment (ConnorVG) [inline tests/TestListener.php]: :wink:

Comment (yoyosan) [inline tests/TestListener.php]: Wouldn't calling the `artisan config:clear` be easier?

Comment (ConnorVG) [inline tests/TestListener.php]: To go through the trouble of _booting another instance_? We also need to ensure tests don't mutate any other compiled or cached data like `services` and the configs in `cachet/*`.

This is all-round quicker to execute and more exhaustive.

**Pull request merged**